x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/MultiShardTermsEnum.java

```diff
@@ -38,45 +38,50 @@ public final class MultiShardTermsEnum {
 
     private int numTop;
     private BytesRef current;
+    private Function<Object, Object> termsDecoder;
 
-    /** Sole constructor.
+    /**
+     * Sole constructor.
+     *
      * @param enums TermsEnums from shards which we should merge
      * @throws IOException Errors accessing data
      **/
-    public MultiShardTermsEnum(TermsEnum[] enums) throws IOException {
+    public MultiShardTermsEnum(TransportTermsEnumAction.ShardTermsEnum[] enums) throws IOException {
```

Comment:
This might be neater as a builder object, and then ShardTermsEnum can become private to MultiShardTermsEnum.  So the caller is something like:
```
MultiShardTermsEnum.Builder termsBuilder = new MultiShardTermsEnum.Builder();
for (ShardId shardId : request.shardIds()) {
  ...
  terms.add(fieldType.getTerms( ... ), fieldType::valueForDisplay)
...
}
MultiShardTermsEnum terms = termsBuilder.build();
```